File: Signiture-Backend/kotlin/study/src/main/java/compatibility/service/UserService.java
package compatibility.service;

import pcy.study.compatibility.model.KotlinUser;

import java.time.LocalDateTime;

public class UserService {

    private static void createKotlinUser(KotlinUser kotlinUser) {
        var user = new KotlinUser(
                "홍길동",
                20,
                kotlinUser.getEmail(),
                kotlinUser.getPhoneNumber(),
                LocalDateTime.now()
        );

        System.out.println(user);
    }

    public static void main(String[] args) {
        var kotlinUser = new KotlinUser(
                "Dummy",
                0,
                "email@gmail.com",
                "000-0000-0000",
                LocalDateTime.of(2000, 8, 23, 16, 25, 30)
        );

        UserService.createKotlinUser(kotlinUser);
    }
}

File: Signiture-Backend/kotlin/study/src/main/kotlin/compatibility/service/UserService.kt
package pcy.study.compatibility.service

import compatibility.model.JavaUser
import java.time.LocalDateTime

class UserService {

    fun createJavaUser(javaUser: JavaUser? = null) {
        val user = JavaUser(
            "홍길동",
            20,
            javaUser?.email,
            javaUser?.phoneNumber,
            LocalDateTime.now(),
        )

        user.name?.let {
            println(it)
        } ?: println("Is Null")
    }
}

fun main() {
    val javaUser = JavaUser(
        "Dummy",
        0,
        "email@gmail.com",
        "000-0000-0000",
        LocalDateTime.of(2000, 8, 23, 16, 25, 30),
    )

    UserService().createJavaUser(javaUser)
}